### PR TITLE
Bump Promscale extension to 0.5.1

### PR DESCRIPTION
**Pull request by JamesGuthrie**

### CODE REVIEW

**Review by sumerman — APPROVED** [review]

**Comment by feikesteenbergen** [comment]
```
2022-06-09T14:16:42.8084918Z #55 1753.6 cp -a ./target/release/promscale-pg14/* /
2022-06-09T14:16:42.8086017Z #55 1753.6 cp: preserving times for '/usr/share/postgresql/14/extension': Operation not permitted
2022-06-09T14:16:42.8087035Z #55 1753.6 cp: preserving times for '/usr/share/postgresql/14': Operation not permitted
2022-06-09T14:16:42.8088315Z #55 1753.6 cp: preserving times for '/usr/share/postgresql': Operation not permitted
2022-06-09T14:16:42.8089247Z #55 1753.6 cp: preserving times for '/usr/share': Operation not permitted
2022-06-09T14:16:42.9596010Z #55 1753.7 cp: preserving times for '/usr/lib/postgresql/14/lib': Operation not permitted
2022-06-09T14:16:42.9596960Z #55 1753.7 cp: preserving times for '/usr/lib/postgresql/14': Operation not permitted
2022-06-09T14:16:42.9597730Z #55 1753.7 cp: preserving times for '/usr/lib/postgresql': Operation not permitted
2022-06-09T14:16:42.9598458Z #55 1753.7 cp: preserving times for '/usr/lib': Operation not permitted
2022-06-09T14:16:42.9599116Z #55 1753.7 cp: preserving times for '/usr': Operation not permitted
```

This may be because the user compiling at this point is `postgres`.

File: build_scripts/install_promscale.sh [hunk 1]
```diff
@@ -36,7 +37,12 @@ for PROMSCALE_VERSION in "$@"; do
     cargo pgx init --pg${PGVERSION} /usr/lib/postgresql/${PGVERSION}/bin/pg_config
     if [ "${MAJOR}" -le 0 ] && [ "${MINOR}" -le 3 ]; then
         PG_VER=pg${PGVERSION} make install || exit 1;
-    else
+    elif [ "${MAJOR}" -le 0 ] && [ "${MINOR}" -le 5 ] && [ "${PATCH}" -le 0 ]; then
         (make package && make install) || exit 1;
+    else
+        make
```
[the remark this answers is not in the document]
Comment (sumerman) [reply build_scripts/install_promscale.sh:43]: TBH I don't understand why permissions broke. Yes, the `install` target was changed, but it shouldn't have affected permissions 🤔

[the remark this answers is not in the document]
Comment (JamesGuthrie) [reply build_scripts/install_promscale.sh:43]: I think what changed is that we're trying to `cp -a` over `/`, which is messing with existing directories. Before we were using pgx's install command, which I guess just copies individual files into the target dir.

[the remark this answers is not in the document]
Comment (feikesteenbergen) [reply build_scripts/install_promscale.sh:43]: The `cp -a` is interesting, whether on a dev machine or on a CI/CD run, many postgres tools will need to be compiled while running as the `postgres` user.

(The reason is that running a `postgres` instance is not allowed as the `root` user, so anything that needs a database during make check for example will need to be running as postgres?).

The `cp -a`, if run as `root` directly, or with `sudo` would then preserve the `postgres` user as owner of those files, which would surprise me, I would expect if i do:

`sudo make install` that the files copied will receive the permissions from the target dirs.

[the remark this answers is not in the document]
Comment (feikesteenbergen) [reply build_scripts/install_promscale.sh:43]: I would not expect on most machines to have files owned by `postgres` in the `/usr/share/postgresql/14` dirs.

**Comment by JamesGuthrie** [comment]
Closing this in favour of https://github.com/timescale/timescaledb-docker-ha/pull/265 which correctly works around our broken `make install` target.

**Pull request closed without merging**